examples/adversarial_training.ipynb

```diff
@@ -0,0 +1,1052 @@
+        "    ds = ds.shuffle(10 * batch_size, seed=0)\n",
+        "  ds = ds.batch(batch_size)\n",
+        "  return iter(tfds.as_numpy(ds)), ds_info\n",
+        "\n",
+        "\n",
+        "class CNN(nn.Module):\n",
+        "  \"\"\"A simple CNN model.\"\"\"\n",
+        "  num_classes: int\n",
+        "\n",
+        "  @nn.compact\n",
+        "  def __call__(self, x):\n",
+        "    x = nn.Conv(features=32, kernel_size=(3, 3))(x)\n",
+        "    x = nn.relu(x)\n",
+        "    x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))\n",
+        "    x = nn.Conv(features=64, kernel_size=(3, 3))(x)\n",
+        "    x = nn.relu(x)\n",
+        "    x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))\n",
+        "    x = x.reshape((x.shape[0], -1))  # flatten\n",
+        "    x = nn.Dense(features=256)(x)\n",
+        "    x = nn.relu(x)\n",
+        "    x = nn.Dense(features=self.num_classes)(x)\n",
+        "    return x"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 6,
+      "metadata": {
+        "id": "trBo7PpcQ4N8",
+        "outputId": "aa85ee43-aabc-4545-d753-a670bd3dbbf5",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 84,
+          "referenced_widgets": [
+            "e1c2e68ebfde40d0a378d864591454ae",
+            "0809e931a2cf4d79a0b4f4adc74cc0a6",
+            "446a3d8753bc4518afc6931dc7a04036",
+            "0826f6b966d0462b93340f98cd0def17",
+            "0bae2a46b67742c681f9d5d2e438be08",
+            "27ce1b1ca61e44df8de592a60713aa19",
+            "d8eec547f16943c99f3c501fca38a927",
+            "dfb4d7931ae14f11bc47d0d8508daee5",
+            "5d7a7618e370456480654b4f1dfbdda2",
+            "cede9d5af636471ab98ffe3e48a275df",
+            "0d2e55eabc6041479bf88a6fbecff7f4"
+          ]
+        }
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Downloading and preparing dataset 11.06 MiB (download: 11.06 MiB, generated: 21.00 MiB, total: 32.06 MiB) to /root/tensorflow_datasets/mnist/3.0.1...\n"
+          ]
+        },
+        {
+          "output_type": "display_data",
+          "data": {
+            "text/plain": [
+              "Dl Completed...:   0%|          | 0/5 [00:00<?, ? file/s]"
+            ],
+            "application/vnd.jupyter.widget-view+json": {
+              "version_major": 2,
+              "version_minor": 0,
+              "model_id": "e1c2e68ebfde40d0a378d864591454ae"
+            }
+          },
+          "metadata": {}
+        },
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Dataset mnist downloaded and prepared to /root/tensorflow_datasets/mnist/3.0.1. Subsequent calls will reuse this data.\n"
+          ]
+        }
+      ],
+      "source": [
+        "train_ds, ds_info = load_dataset(\"train\", is_training=True,\n",
```

Comment:
please use a simpler loader like the one in https://optax.readthedocs.io/en/latest/_collections/examples/mlp_mnist.html that doesn't define its own load_dataset function. I believe also this might help fix the test errors we're getting